curp/src/server/bg_tasks.rs

```diff
@@ -1,107 +1,86 @@
-use std::{
-    collections::{HashMap, HashSet},
-    iter,
-    sync::{atomic::Ordering, Arc},
-    time::Duration,
-};
+use std::{collections::HashMap, sync::Arc, time::Duration};
 
 use clippy_utilities::NumericCast;
 use event_listener::Event;
-use futures::{future::Either, pin_mut, stream::FuturesUnordered, Stream, StreamExt};
-use itertools::Itertools;
+use futures::{pin_mut, stream::FuturesUnordered, Stream, StreamExt};
 use madsim::rand::{thread_rng, Rng};
-use parking_lot::{
-    lock_api::{RwLockUpgradableReadGuard, RwLockWriteGuard},
-    RawRwLock,
-};
-use tokio::{sync::mpsc, time::MissedTickBehavior};
+use tokio::{sync::mpsc, task::JoinHandle, time::MissedTickBehavior};
 use tracing::{debug, error, info, warn};
-use utils::{
-    config::ServerTimeout,
-    parking_lot_lock::{MutexMap, RwLockMap},
-};
 
-use super::{state::StateRef, SyncMessage};
+use super::{
+    raw_curp::{AppendEntries, RawCurp, TickAction, Vote},
+    SyncMessage,
+};
 use crate::{
-    cmd::{Command, ProposeId},
+    cmd::Command,
     log::LogEntry,
+    message::ServerId,
     rpc::{
-        self, connect::ConnectInterface, AppendEntriesRequest, AppendEntriesResponse, VoteRequest,
-        VoteResponse,
+        self,
+        connect::{Connect, ConnectInterface},
+        AppendEntriesRequest, AppendEntriesResponse, VoteRequest, VoteResponse,
     },
-    server::{cmd_worker::CmdExeSenderInterface, ServerRole, State},
     TxFilter,
 };
 
+/// Connects
+type Connects<Conn> = HashMap<ServerId, Arc<Conn>>;
+
 /// Run background tasks
 #[allow(clippy::too_many_arguments)] // we call this function once, it's ok
-pub(super) async fn run_bg_tasks<
-    C: Command + 'static,
-    Conn: ConnectInterface,
-    ExeTx: CmdExeSenderInterface<C>,
->(
-    state: StateRef<C, ExeTx>,
-    sync_rx: flume::Receiver<SyncMessage<C>>,
-    cmd_exe_tx: ExeTx,
+pub(super) async fn run_bg_tasks<C: Command + 'static>(
+    curp: Arc<RawCurp<C>>,
+    others: HashMap<ServerId, String>,
+    sync_chan: flume::Receiver<SyncMessage<C>>,
```

Comment:
I think `sync_rx` is more straightforward than `sync_chan` in that the former indicates that it is the reveive half of the channel.

Reply:
Fixed. Bg tasks will soon be refactored by the next PR.